app/src/main/java/com/example/tasky/authentication/domain/PasswordValidator.kt

```diff
@@ -0,0 +1,11 @@
+package com.example.tasky.authentication.domain
+
+import javax.inject.Inject
+
+class PasswordValidator @Inject constructor() {
+
+    private val passwordRegex = Regex("^(?=.*\\d)(?=.*[a-z])(?=.*[A-Z]).{9,}$")
+    fun validatePassword(password: String): Boolean {
+        return password.matches(passwordRegex)
+    }
```

Comment:
According to Tasky´s specifications
We want to let the user know that they are missing a number, lowercase, uppercase, or if their password isn´t long enough

You can´t do that using Regex

So for this case, using multiple if statements would be the way to go

Reply:
![image](https://github.com/MelvinKucuk/tasky/assets/42274125/4cdd3df7-ec16-4c1d-872e-82766f9a6c7b)
I didnt saw that on the requirements but if it is that way I would do it that way

Reply:
Oh nevermind. The specs are different now. It's okay that way then

Personally, I still don't like regexes since they are hard to understand without proper testing
But if you like it, that's fine